downgrade

```diff
@@ -103,8 +113,8 @@ main() {
   for term; do
     candidates=( $(search_packages "$term" | sort_packages) )
 
-    if present_packages "${candidates[@]}"; then
-      if choice="$(read_selection "${candidates[@]}")"; then
+    if check_packages || test "${#candidates[@]}" -gt 0; then
+      if check_selection || choice="${candidates[0]}"; then
```

Comment:
I'm having a little trouble following the intent of these nested, combined conditionals now. Can you explain what this code is doing? Then I might have some suggestions for clarifying the naming.